### ISSUE TITLE
Support SSL Client Cert Auth for replica endpoints

### ISSUE THREAD

**Issue by coolacid** [goharbor/harbor#7176]
I currently have a docker registry deployed in public space, for this registry, an nginx server sits in front of the registry providing SSL Client Certificate authentication. 

I don't want to run a full instance of harbor in public space, and I'd prefer to keep the SSL Client Cert authentication. Thus, I'm hoping to have our CI systems push images into a local non-public instance of Harbor, which would then replicate into our public space registry. 

This would require Harbor supporting more authentication methods then just User/Password. 

I envision a way to either copy/paste or upload a certificate bundle that can be used to authenticate against the remote registry within the add endpoint 

Additional details on how docker handles this: https://docs.docker.com/engine/security/certificates/

**Comment by reasonerjt** [comment]
@coolacid I think the doc in your description does not really handle authentication.  Could you let us know a registry that handles authentication via bi-directional tls?

**Comment by coolacid** [comment]
The doc in the description shows how to add SSL CA and client certs to the client dockerd installation. 

Docker registry doesn't provide SSL client certificate authentication out of the box and required a reverse proxy. We used nginx; IE: https://docs.nginx.com/nginx/admin-guide/security-controls/securing-http-traffic-upstream/

**Comment by reasonerjt** [comment]
Given it's not a very common practice to authenticate registry using TLS cert, mark it as a low priority for now, once the replication refactor is done, we'll revisit.

**Comment by coolacid** [comment]
Thanks - Would there be a simple way to manually add SSL Certs using docker volumes as a work around?

**Comment by shawngmc** [comment]
I would be interested in this feature as well.

Docker Trusted Registry supports this feature: https://docs.docker.com/ee/enable-client-certificate-authentication/

@coolacid Do you have any documentation on the Nginx config you set up to pass the cert DN forward to harbor core and/or the token service (for use with the CLI)?

**Comment by germaino** [comment]
I have done a POC  to support certificate based TLS mutual authentication here: https://github.com/germaino/harbor/tree/germaino. I am able to successfully authenticate with a client certificate and pull some image manifests with the docker API

**Comment by 0x217** [comment]
@coolacid Can you share the configuration of Nginx with the client certificate method?

**Comment by Vad1mo** [comment]
Triage note: this looks like a duplicate of #22558, which tracks the same topic (mTLS/client cert to external registries). Suggest consolidating the discussion there — please follow #22558 for updates. Adding the `duplicate` label and leaving this open for maintainer confirmation.
